sage-ios/SAGE/SAGE/AdminOperations.swift

```diff
@@ -115,44 +146,126 @@ class AdminOperations {
+        }
     }
     
     static func editSchool(school: School, completion: ((School) -> Void)?, failure: (String) -> Void){
         let manager = BaseOperation.manager()
-        // TODO: make a network request
+        
+        let params = ["school": [
+                SchoolConstants.kLat: school.location!.coordinate.latitude,
+                SchoolConstants.kLong: school.location!.coordinate.longitude,
+                SchoolConstants.kName: school.name!,
+                SchoolConstants.kDirectorID: school.director!.id
+            ]
+        ]
+        let schoolURLString = StringConstants.kSchoolAdminDetailURL(school.id)
+        
+        manager.PATCH(schoolURLString, parameters: params, success: { (operation, data) -> Void in
+            completion!(school)
+            }) { (operation, error) -> Void in
+                failure("Could not edit school.")
+        }
     }
     
     static func editAnnouncement(announcement: Announcement, completion: ((Announcement) -> Void)?, failure: (String) -> Void){
         let manager = BaseOperation.manager()
-        // TODO: make a network request
+        let params = [
+                AnnouncementConstants.kTitle: announcement.title!,
+                AnnouncementConstants.kText: announcement.text!,
+                AnnouncementConstants.kSchoolID: announcement.school!.id,
+        ]
+        let announcementURLString = StringConstants.kAnnouncementAdminDetailURL(announcement.id!)
+        manager.PATCH(announcementURLString, parameters: params, success: { (operation, data) -> Void in
+            completion!(announcement)
+            }) { (operation, error) -> Void in
+                failure("Could not edit announcement.")
+        }
     }
     
     static func approveCheckin(checkin: Checkin, completion: (() -> Void)?, failure: (String) -> Void) {
         let manager = BaseOperation.manager()
-        // TODO: make a network request
+        
+        let checkinURLString = StringConstants.kCheckinAdminVerifyURL(checkin.id)
+        
+        manager.POST(checkinURLString, parameters: nil, success: { (operation, data) -> Void in
+            completion?()
+            }) { (operation, error) -> Void in
+                failure("Could not approve checkin.")
+        }
     }
     
     static func removeCheckin(checkin: Checkin, completion: (() -> Void)?, failure: (String) -> Void) {
         let manager = BaseOperation.manager()
-        // TODO: make a network request
+        let checkinURLString = StringConstants.kCheckinAdminDetailURL(checkin.id)
+        
+        manager.DELETE(checkinURLString, parameters: nil, success: { (operation, data) -> Void in
+            completion?()
+            }) { (operation, error) -> Void in
+                failure("Could not remove checkin.")
+        }
+
     }
     
     static func verifyUser(user: User, completion: (() -> Void)?, failure: (String) -> Void) {
+        // TODO: fix, currently doesn't verify
```

Comment:
the rails endpoint didnt exist to verify users before, now it does so i'll get rid of this